packages/components/src/Combobox/components/ComboboxActivator/ComboboxActivator.tsx

```diff
@@ -1,24 +1,19 @@
 import React from "react";
-import { Button } from "@jobber/components/Button";
-import { Chip } from "@jobber/components/Chip";
 import { ComboboxContext } from "../../ComboboxProvider";
 import { ComboboxActivatorProps } from "../../Combobox.types";
 
 export function ComboboxActivator(props: ComboboxActivatorProps) {
-  const { handleClose, open, setOpen } = React.useContext(ComboboxContext);
+  const { handleClose, open, handleOpen } = React.useContext(ComboboxContext);
 
-  if (props.children.type === Button || props.children.type === Chip) {
```

Comment:
a few reasons I want to try doing it this was VS a children render prop, or a named render prop - despite us wanting to use that same format as often as we can

**Reduce Responsibility**

if we take in whatever markup someone wants to provide and then clone it along with the enhancements of aria attributes and the `onClick` needed for the activator - then the consumer doesn't have to go read our docs about what's required of a combobox as far as accessibility.

also, if for some reason we want to change the behavior on click - since that's still under our control, a consumer wouldn't have to update their implementation to match.

all the consumer needs to do is provide the element, and if they wanted to - a separate clickable/interactive element that they'd have to stop event propagation inside so as to not trigger the open/close

**Avoid multiple ways of doing the same thing**

since Combobox and this Activator are some of the earlier usages of compound components and (limited) composition/flexibility - we already have the API for deciding to provide your own markup

if you want the default. don't add a `Combobox.Activator`. 

```
<Combobox {...props}>
  // Options and Actions
</Combobox>
```

if you do, provide one with either a `Chip` or `Button` as the immediate child

```
<Combobox {...props}>
  <Combobox.Activator>
    <Button label="whatever"/>
  </Combobox.Activator>
  // Options and Actions
</Combobox>
```

we currently have 24 instances of custom activators with this pattern and around 11 that use the default.

with that in mind, if we do the `customRenderItem` that's 2 ways of providing your own markup. if it's a Chip or Button, use the children. if it's something else, use the prop.

if we do the children render function, that's a bit better since it's more similar however it's still 2 ways. if it's a Chip or Button, provide markup. if it's something else, provide a function and implement the accessibility, and open/close behavior yourself.

```
 <Combobox.Activator>
   <Chip variant="subtle"/>
 </Combobox.Activator/>
 
 or
 
 <Combobox.Activator>
   {{...activatorAPI} => (<div 
   onClick={activatorAPI.onClick} 
   role={activatorAPI.role} 
   aria-expanded={activatorAPI.open}>
     <Avatar url="smile.jpg" />
   </div>)}
 </Combobox.Activator>
```

as I mentioned above - this is a lot of responsibility for a consumer. the counter point there is that it's more power, but I'm not convinced anyone really wants the power to change the behavior of the Activator nor is that necessarily a place for us to be un-opinionated.

I think it's fine for us to say, when you click on this element it toggles the combobox's visibility. that is the exact purpose and function of this component. how it looks though, is up to you.

Reply:
Seems reasonable

Reply:
I think this seems reasonable. I feel like always using a named render prop vs `children` will require us to have exceptions. Also another reason to use `children` in this case is that `Combobox.Activator` is explicitly how we allow consumers to define a custom activator (you can't use it without passing `children`). 

If people want the default activator they wouldn't pass a `Combobox.Activator` into into the`Combobox`. In the future we _could_ remove `Combobox.Activator` and move it into a named prop on the `Combobox` but I think that is not a good idea, especially right now